zipline/algorithm.py

```diff
@@ -613,12 +606,9 @@ def run(self, data=None, overwrite_sim_params=True):
                 # For compatibility with existing examples allow start/end
                 # to be inferred.
                 if overwrite_sim_params:
-                    self.sim_params.period_start = data.major_axis[0]
-                    self.sim_params.period_end = data.major_axis[-1]
-                    # Changing period_start and period_close might require
-                    # updating of first_open and last_close.
-                    self.sim_params.update_internal_from_trading_schedule(
-                        trading_schedule=self.trading_schedule
+                    self.sim_params = self.sim_params.create_from_copy(
```

Comment:
I renamed it to `create_new`.  Meant to take a start and end date, and return a new `SimulationParameters` created from an existing `SimulationParameters`, with those fields overwritten.